File: database/src/main/java/de/reiss/bible2net/theword/database/NoteItemDao.java
package de.reiss.bible2net.theword.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.Date;
import java.util.List;

import static androidx.room.OnConflictStrategy.REPLACE;

@Dao
public interface NoteItemDao {

    @Query("SELECT * FROM NoteItem")
    List<NoteItem> all();

    @Query("SELECT * FROM NoteItem WHERE date = :date")
    NoteItem byDate(Date date);

    @Query("SELECT * FROM NoteItem WHERE date BETWEEN :from AND :to")
    List<NoteItem> range(Date from, Date to);

    @Insert(onConflict = REPLACE)
    List<Long> insertOrReplace(NoteItem... items);

    @Delete
    int delete(NoteItem... item);

    @Query("DELETE FROM NoteItem")
    void clear();

}

File: app/src/main/java/de/reiss/bible2net/theword2/database/NoteItemDao.kt
package de.reiss.bible2net.theword2.database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import java.util.Date

@Dao
interface NoteItemDao {

    @Query("SELECT * FROM NoteItem")
    fun all(): List<NoteItem>

    @Query("SELECT * FROM NoteItem WHERE date = :date")
    fun byDate(date: Date): NoteItem?

    @Query("SELECT * FROM NoteItem WHERE date BETWEEN :from AND :to")
    fun range(from: Date, to: Date): List<NoteItem>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertOrReplace(vararg items: NoteItem): List<Long>

    @Delete
    fun delete(vararg item: NoteItem): Int

    @Query("DELETE FROM NoteItem")
    fun clear()
}
